crates/oxc_transformer/src/plugins/module_runner_transform.rs

```diff
@@ -0,0 +1,2055 @@
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue');
+Object.defineProperty(__vite_ssr_exports__, 'foo', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return __vite_ssr_import_0__;
+  }
+});",
+        );
+    }
+
+    #[test]
+    fn re_export_by_imported_name() {
+        test_same(
+            "import * as foo from 'foo'; export * as foo from 'foo'",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('foo');
+const __vite_ssr_import_1__ = await __vite_ssr_import__('foo');
+Object.defineProperty(__vite_ssr_exports__, 'foo', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return __vite_ssr_import_1__;
+  }
+});",
+        );
+
+        test_same(
+            "import { foo } from 'foo'; export { foo } from 'foo'",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('foo', { importedNames: ['foo'] });
+const __vite_ssr_import_1__ = await __vite_ssr_import__('foo', { importedNames: ['foo'] });
+Object.defineProperty(__vite_ssr_exports__, 'foo', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return __vite_ssr_import_1__.foo;
+  }
+});",
+        );
+    }
+
+    #[test]
+    fn export_arbitrary_namespace() {
+        test_same(
+            "export * as \"arbitrary string\" from 'vue'",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue');
+Object.defineProperty(__vite_ssr_exports__, 'arbitrary string', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return __vite_ssr_import_0__;
+  }
+});",
+        );
+
+        test_same(
+            "const something = \"Something\"; export { something as \"arbitrary string\" }",
+            "const something = 'Something';
+Object.defineProperty(__vite_ssr_exports__, 'arbitrary string', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return something;
+  }
+});",
+        );
+
+        test_same(
+            "export { \"arbitrary string2\" as \"arbitrary string\" } from 'vue'",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['arbitrary string2'] });
+Object.defineProperty(__vite_ssr_exports__, 'arbitrary string', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return __vite_ssr_import_0__['arbitrary string2'];
+  }
+});",
+        );
+    }
+
+    #[test]
+    fn export_default() {
+        test_same(
+            "export default {}",
+            "Object.defineProperty(__vite_ssr_exports__, 'default', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return {};
+  }
+});",
+        );
+    }
+
+    #[test]
+    fn export_then_import_minified() {
+        test_same(
+            "export * from 'vue';import {createApp} from 'vue'",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue');
+__vite_ssr_exportAll__(__vite_ssr_import_0__);
+const __vite_ssr_import_1__ = await __vite_ssr_import__('vue', { importedNames: ['createApp'] });",
```

Comment:
The order is different from Vite.
https://github.com/vitejs/vite/blob/2c51565ec044904a080ef5649034c37f02212c7b/packages/vite/src/node/ssr/__tests__/ssrTransform.spec.ts#L211
But it should be fine. (just pointing out for future reference)

Reply:
Thank you for pointing it out.